File: WyrmBook/WyrmBook.Core/Models/DragonModel.cs
using WyrmBook.Domain.Entities;
using WyrmBook.Domain.Enums;

namespace WyrmBook.Core.Models;

public record DragonModel
{
    public required int Age { get; init; }
    public required string Name { get; init; }
    public required ElementalType ElementalType { get; init; }
    public required DragonSubType SubType { get; init; }
}

File: WyrmBookFrontend/src/data/models/dragon.model.ts
enum ElementalType {
  // Define the enum values according to WyrmBook.Domain.Enums.ElementalType
  // Example:
  Fire,
  Water,
  Earth,
  Air,
}

enum DragonSubType {
  // Define the enum values according to WyrmBook.Domain.Enums.DragonSubType
  // Example:
  Western,
  Eastern,
  Wyvern,
  Drake,
}

interface Dragon {
  /**
   * Mystical age of the dragon
   */
  age: number;

  /**
   * Mystical name of the dragon
   */
  name: string;

  /**
   * Mystical dragon elemental type
   */
  elementalType: ElementalType;

  /**
   * Mystical dragon subtype
   */
  subType: DragonSubType;

  /**
   * Amount of legs the dragon-type has
   */
  amountOfLegs: number;

  /**
   * Amount of wings the dragon-type has
   */
  amountOfWings: number;
}

export { Dragon, ElementalType, DragonSubType };
